crates/egui/src/containers/scroll_area.rs

```diff
@@ -779,15 +779,33 @@ impl Prepared {
         if scrolling_enabled && is_hovering_outer_rect {
             for d in 0..2 {
                 if scroll_enabled[d] {
-                    let scroll_delta = ui.ctx().frame_state(|fs| fs.scroll_delta);
+                    let scroll_delta = ui.ctx().frame_state(|fs| {
+                        if ui.style().always_scroll_the_only_direction
+                            && scroll_enabled[0] != scroll_enabled[1]
```

Comment:
`ui.style().always_scroll_the_only_direction && scroll_enabled[0] != scroll_enabled[1]` is repeated down below. It would be nice to put it in a `bool` and reuse it to make the logic more clear; though I don't know what name to give it…